code/src/index.css

```diff
@@ -1,13 +1,94 @@
+* {
+  /* border: 1px red solid; */
+}
+
 body {
   margin: 0;
   font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
     "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background: #181818;
 }
 
 code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
+.icon-container {
+  position: absolute;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+  opacity: 0;
+  transition: 0.4s;
+  height: 100%;
+}
+
+.icon-container:hover {
+  opacity: 1;
+  background-color: rgb(0 0 0 / 43%);
+}
+
+.icon {
+  width: 50px;
+  margin: 4px;
+  filter: invert(1);
```

Comment:
Having a 1 in the invert also makes the icon fully white. If you lower the 1 to for example 0.5 you will see the color change to grey-white. It will then go to white in your hover state on line 33 as you have written opacity 1 there.